drv/stm32xx-i2c-server/src/main.rs

```diff
@@ -444,6 +445,66 @@ fn configure_port(
     map.insert(controller.controller, port);
 }
 
+///
+/// When the system is reset without power loss, I2C can be in an arbitrary
+/// state with respect to the bus -- and we can therefore come to life with a
+/// transaction already in flight.  It is very important that we abort any
+/// such transaction:  failure to do so will result in our first I2C
+/// transaction being corrupted.  (And especially because our first I2C
+/// transactions may well be to disable segments on a mux, this can result in
+/// nearly arbitrary mayhem down the road!)  To do this, we engage in the
+/// time-honored[0] tradition of "clocking through the problem":  wiggling SCL
+/// until we see SDA high, and then pulling SDA low and releasing SCL to
+/// indicate a STOP condition.  (Note that we need to do this up to 9 times to
+/// assure that we have clocked through the entire transaction.)
+///
+/// [0] Analog Devices. AN-686: Implementing an I2C Reset. 2003.
+///
+fn wiggle_scl(sys: &Sys, scl: PinSet, sda: PinSet) {
+    sys.gpio_configure_input(sda, Pull::None);
+
+    sys.gpio_configure_output(
```

Comment:
We may want to set this GPIO to high _before_ configuring it as an output; otherwise, I think it will pull the clock low immediately.

Reply:
Done -- thank you!